src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Utilities/SyntaxKindSet.cs

```diff
@@ -93,6 +93,12 @@ internal class SyntaxKindSet
             SyntaxKind.RecordDeclaration,
         };
 
+        public static readonly ISet<SyntaxKind> ClassRecordTypeDeclarations = new HashSet<SyntaxKind>(SyntaxFacts.EqualityComparer)
+        {
+            SyntaxKind.ClassDeclaration,
+            SyntaxKind.RecordDeclaration,
+        };
```

Comment:
> Is RecordStructDeclaration intentionally missing here? 

This is a clone of ```ClassInterfaceRecordTypeDeclarations``` without interface declaration. This is the difference I wanted at the use site. I think ```RecordStructDeclaration``` is intentionally missing from ```ClassInterfaceRecordTypeDeclarations``` because it is used for abstract/sealed recommender and those are never allowed in structures. 

> Is there any test showing the difference between a record and a record struct?

Added/modified tests cover all flavors of records and demonstrate no difference in behavior between them.